kuma/wiki/forms.py

```diff
@@ -233,47 +250,55 @@ def __init__(self, *args, **kwargs):
         if self.section_id:
             self.fields['toc_depth'].required = False
 
-    def _clean_collidable(self, name):
-        value = self.cleaned_data[name]
-
+    def clean_slug(self):
+        # Since this form can change the URL of the page on which the editing
+        # happens, changes to the slug are ignored for an iframe submissions
         if self.is_iframe_target:
-            # Since these collidables can change the URL of the page, changes
-            # to them are ignored for an iframe submission
-            return getattr(self.instance.document, name)
+            return self.instance.document.slug
 
-        error_message = {'slug': SLUG_COLLIDES}.get(name, OTHER_COLLIDES)
+        # first check if the given slug doesn't contain slashes and other
+        # charaters not allowed in a revision slug component (without parent)
+        if (self.cleaned_data['slug'] and
+                INVALID_REV_SLUG_CHARS_RE.search(self.cleaned_data['slug'])):
+            raise forms.ValidationError(SLUG_INVALID)
+
+        # edits can come in without a slug, so default to the current doc slug
         try:
-            existing_doc = Document.objects.get(
-                locale=self.instance.document.locale,
-                **{name: value})
+            if not self.cleaned_data['slug']:
+                document_slug = self.instance.document.slug
+                self.cleaned_data['slug'] = self.instance.slug = document_slug
+        except ObjectDoesNotExist:
```

Comment:
`self.instance.document` may raise a `RelatedObjectDoesNotExist` which is constructed at runtime and I'd rather catch the base class: https://github.com/django/django/blob/1bbca7961cee20c4ddd453a7d74d316e84f4bbb5/django/db/models/fields/related.py#L433-L441